src/pkg/utils/io.go

```diff
@@ -327,6 +328,140 @@ func SplitFile(path string, chunkSizeBytes int) (chunks [][]byte, sha256sum stri
+// - sha256sum: sha256sum of the srcFile before splitting
+// - err: any errors encountered
+func SplitFile(srcFile string, chunkSizeBytes int) (err error) {
+	var fileNames []string
+	var sha256sum string
+	hash := sha256.New()
+
+	// Set buffer size to some multiple of 4096 KiB for modern file system cluster sizes
+	bufferSize := 16 * 1024 * 1024 // 16 MiB
+	// if chunkSizeBytes is less than bufferSize, use chunkSizeBytes as bufferSize for simplicity
+	if chunkSizeBytes < bufferSize {
+		bufferSize = chunkSizeBytes
+	}
+	buf := make([]byte, bufferSize)
+
+	// get file size
+	fi, err := os.Stat(srcFile)
+	if err != nil {
+		return err
+	}
+	fileSize := fi.Size()
+
+	// start progress bar
+	title := fmt.Sprintf("[0/%d] MB bytes written", fileSize/1000/1000)
+	progressBar := message.NewProgressBar(fileSize, title)
+	defer progressBar.Stop()
+
+	// open file
+	file, err := os.Open(srcFile)
+	defer file.Close()
+	if err != nil {
+		return err
+	}
+
+	// create file path starting from part 001
+	path := fmt.Sprintf("%s.part001", srcFile)
+	chunkFile, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	fileNames = append(fileNames, path)
+	defer chunkFile.Close()
+
+	// setup counter for tracking how many bytes are left to write to file
+	chunkBytesRemaining := chunkSizeBytes
+	// Loop over the tarball hashing as we go and breaking it into chunks based on the chunkSizeBytes
+	for {
+		bytesRead, err := file.Read(buf)
+
+		if err != nil {
+			if err == io.EOF {
+				// At end of file, break out of loop
+				break
+			}
+			return err
+		}
+
+		// Pass data to hash
+		hash.Write(buf[0:bytesRead])
+
+		// handle if we should split the data between two chunks
+		if chunkBytesRemaining < bytesRead {
+			// write the remaining chunk size to file
+			_, err := chunkFile.Write(buf[0:chunkBytesRemaining])
+			if err != nil {
+				return err
+			}
+
+			// create new file
+			path = fmt.Sprintf("%s.part%03d", srcFile, len(fileNames)+1)
+			chunkFile, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY, 0644)
+			if err != nil {
+				return err
+			}
+			fileNames = append(fileNames, path)
+			defer chunkFile.Close()
+
+			// write to new file where we left off
+			_, err = chunkFile.Write(buf[chunkBytesRemaining:bytesRead])
+			if err != nil {
+				return err
+			}
+
+			// set chunkBytesRemaining considering how many bytes are already written to new file
+			chunkBytesRemaining = chunkSizeBytes - (bufferSize - chunkBytesRemaining)
+		} else {
+			_, err := chunkFile.Write(buf[0:bytesRead])
+			if err != nil {
+				return err
+			}
+			chunkBytesRemaining = chunkBytesRemaining - bytesRead
+		}
+
+		// update progress bar
+		progressBar.Add(bufferSize)
+		title := fmt.Sprintf("[%d/%d] MB bytes written", progressBar.GetCurrent()/1000/1000, fileSize/1000/1000)
+		progressBar.UpdateTitle(title)
+	}
+	progressBar.Successf("Package split across %d files", len(fileNames))
```

Comment:
```suggestion
	progressBar.Successf("Package split across %d files", len(fileNames)+1)
```
Tiny change since the number of files will actually be one higher with the metadata file (on larger numbers people aren't likely to notice but on 2 or 3 they might.

It may also be better to move this to after the metadata file being written as well so it calls success once the whole splitting operation is done.

Reply:
That's a good point. I've moved the progressBar.Successf down a few lines after the metadata file is written. Since the metadata file gets added to `fileNames`, I chose to keep `len(fileNames)` as it should be an accurate count now